test/suites/storage_profiles.sh

```diff
@@ -126,6 +126,25 @@ test_storage_profiles() {
       ! lxc profile assign c"${i}" testDup,testNoDup || false
     done
 
+    # Create a new profile and volume for testing custom block volume sharing.
+    lxc profile create volumeSharingTest
+    lxc storage volume create "lxdtest-$(basename "${LXD_DIR}")-pool1" block-vol --type=block
+
+    # Test adding a non-shared block volume device to a profile. That operation must fail.
+    ! lxc profile device add volumeSharingTest test-disk disk pool="lxdtest-$(basename "${LXD_DIR}")-pool1" source=block-vol || false
+
+    # Then enabling sharing the block volume and trying again, must succeed this time.
+    lxc storage volume set "lxdtest-$(basename "${LXD_DIR}")-pool1" block-vol security.shared true
+    lxc profile device add volumeSharingTest test-disk disk pool="lxdtest-$(basename "${LXD_DIR}")-pool1" source=block-vol
+
+    # Try to disable security.shared for a volume already added to a profile. That operation must fail.
+    ! lxc storage volume set "lxdtest-$(basename "${LXD_DIR}")-pool1" block-vol security.shared false || false
```

Comment:
Should there be another check of this kind if the volume is directly attached to an instance? In the commits before I saw you have added code for this too.

Reply:
yes, I added these these on `lxd-ci` because block volumes can only be attached to VMs and `security.shared` is exclusive to block volumes

Reply:
My thinking was if there is a test checking if `security.shared` cannot be unset if the volume is already attached to a profile, shouldn't there also be a test checking if `security.shared` cannot be unset if the volume is attached to multiple VMs? 

I am talking about this part of the code https://github.com/canonical/lxd/pull/13183/files#diff-029fdd0e1afdacb0e66630c0d7e813772122db9a7d036060902838db315d9c4dR5625

Reply:
I believe [this](https://github.com/canonical/lxd-ci/blob/03117148afa374a8974dc92780621d7ad21f6235/tests/storage-volumes-vm#L106) is the check you are referring to. This is part of https://github.com/canonical/lxd-ci/pull/146. Please let me know if I am understanfing something wrong